### PR TITLE
Preserve whitespace between words in markdown link text (#628)

### PR DESCRIPTION
**Pull request by egnha**
Implements the patch proposed in issue #628.

### CODE REVIEW

File: R/markdown.R [hunk 1]
```diff
@@ -182,7 +182,7 @@ markdown_tags <- list(
     } else if (dest == "" || dest == xml_text(xml)) {
       list("\\url{", xml_text(xml), "}")
     } else {
-      list("\\href{", dest, "}{", xml_text(xml), "}")
+      list("\\href{", dest, "}{", xml_link_text(xml), "}")
```
Comment (gaborcsardi) [inline R/markdown.R]: Can you please call it `xml_href_text` ?

**Comment by gaborcsardi** [comment]
Thanks much, it looks great!

File: R/markdown.R (hunk 1 above)
Comment (egnha) [inline R/markdown.R]: Sure thing.

**Comment by egnha** [comment]
My pleasure. :)

**Comment by gaborcsardi** [comment]
Actually, I wonder if we should do this everywhere. Converting a line break to a space seems like a good default behavior. 

(You don't need to modify this PR, just thinking aloud.)

**Comment by hadley** [comment]
Could you please add a bullet to NEWS?

**Comment by hadley** [comment]
Thanks!

For future reference, please make sure that one of the commit messages includes `Fixes #628` so the accompanying issues gets automatically closed when the PR is merged.

**Comment by egnha** [comment]
I'll keep that point in mind for next time. Thanks for the review and merge!

**Pull request merged**